lib/cloudregister/registerutils.py

```diff
@@ -516,6 +517,63 @@ def get_credentials(credentials_file):
     return (username, password)
 
 
+# ----------------------------------------------------------------------------
+def setup_registry(registry_fqdn, username, password):
+    """Set all the necessary parts for the registry,
+       returns True if the setup completed, False otherwise."""
+    if not os.path.exists(os.path.dirname(REGISTRY_CREDENTIALS_PATH)):
+        os.makedirs(os.path.dirname(REGISTRY_CREDENTIALS_PATH))
+
+    setup_registry_succeed = set_registry_auth_token(
+        registry_fqdn, username, password
+    )
+    return setup_registry_succeed
+
+
+# ----------------------------------------------------------------------------
+def set_registry_auth_token(registry_fqdn, username, password):
+    """Set the auth token to pull images from SUSE registry."""
+    auth_token = base64.b64encode('{username}:{password}'.format(
+        username=username,
+        password=password
+    ).encode()).decode()
+    registry_credentials = {'auths': {registry_fqdn: {'auth': auth_token}}}
+    config_json = {}
+    if os.path.exists(REGISTRY_CREDENTIALS_PATH):
+        try:
+            with open(REGISTRY_CREDENTIALS_PATH, 'r') as cred_json:
+                config_json = json.load(cred_json)
+        except json.decoder.JSONDecodeError:
+            logging.info(
+                'Unable to parse existing %s, preserving file as %s.bak, '
+                'writing new credentials' % (
+                    REGISTRY_CREDENTIALS_PATH, REGISTRY_CREDENTIALS_PATH
+                )
+            )
+            mv_file_cmd = 'mv -Z {} {}.bak'.format(
+                REGISTRY_CREDENTIALS_PATH, REGISTRY_CREDENTIALS_PATH
+            )
+            exec_subprocess(mv_file_cmd.split())
```

Comment:
do we want to log that the preservation of the problematic file suceeded ?
something like
```python
failed = exec_subprocess(mv_file_cmd.split())
message = 'File not created' if failed else 'File created'
logging.info(message)
```

Reply:
Good point. Also reminds me that the logic is not yet complete. If the file exists we should comprehend the file. It could be that the user created a golden image without cleaning up, in which case we would preserve a file with old credentials and nothings else, which we do not want to do. We only want to preserve a file that has content other than the authentication setting for our registry. Meaning we need to do a bit more checking.

Rest looks much better now, thanks

Reply:
Thanks for the feedback

> we need to do a bit more checking

not sure we can ? we try to open the json file and it crashes when parsing with `JSONDecodeError`
how can we check its content ?

Reply:
do you want the code to run `grep` or another command to check its content without accessing it from Python ?